### PR TITLE
Web UI 599 10.10 ondemand pipeline

### PR DESCRIPTION
**Pull request by richardsd**
Initial work on the 10.10 ondemand pipeline (single job alternative to https://github.com/nuxeo/nuxeo-web-ui/pull/1359).

Feedback is very welcome 🙂

### CODE REVIEW

File: .github/actions/get-branch/action.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    required: true
+  branch:
+    description: 'branch name'
+    required: true
+  default-branch:
+    description: 'default branch name'
+    required: true
+outputs:
+  branch:
+    description: "The name of the branch to checkout"
```
Comment (mnixo) [inline .github/actions/get-branch/action.yaml]: In these `.yaml` files, when are we supposed to use `""` or `''`?

**Review by mnixo — DISMISSED** [review]

File: .github/workflows/ondemand-main.yaml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+    - name: Determine nuxeo-elements branch to use
+      id: pick_nuxeo_elements_branch
+      uses: ./.github/actions/get-branch
+      with:
+        repository: nuxeo/nuxeo-elements
+        branch: ${{ github.event.inputs.branch_name }}
+        default-branch: ${{ env.ELEMENTS_REFERENCE_BRANCH }}
+
+    - name: Determine nuxeo-ui-elements branch to use
```
Comment (Gabez0r) [inline .github/workflows/ondemand-main.yaml:86]: We are missing [nuxeo/nuxeo-dataviz-elements](https://github.com/nuxeo/nuxeo-dataviz-elements/tree/maintenance-2.4.x). I see that this is also the case for the current legacy pipeline, so I'd be ok with having a dedicated ticket for this if it causes too much burden.

File: .github/actions/get-branch/action.yaml (hunk 1 above)
Comment (Gabez0r) [reply .github/actions/get-branch/action.yaml]: Indeed, we see to be mixing both.

File: .github/workflows/ondemand-main.yaml [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      run: |
+        pushd nuxeo-web-ui
+        sed -i 's/10.10-HF[0-9]\{2\}-SNAPSHOT/10.10-HF${{ github.event.inputs.hf_version }}/g' pom.xml
+        popd
+
+    - name: Nuxeo package build
+      run: |
+        pushd nuxeo-web-ui
+        mvn install -DskipInstall
+        popd
```
Comment (Gabez0r) [inline .github/workflows/ondemand-main.yaml]: We are repeatedly doing `pushd nuxeo-web-ui` and `popd` on these steps. Do we really need it?

File: .github/workflows/ondemand-main.yaml [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+      with:
+        repository: nuxeo/nuxeo-web-ui-itests
+        path: nuxeo-web-ui-itests
+        ref:  ${{ steps.pick_plugin_nuxeo_web_ui_itests_branch.outputs.branch }}
+
+    # HACK - this is a workaround that will be removed. Necessary because SNAPSHOT version might not be published yet
+    - name: Update nuxeo-web-ui-itests SNAPSHOT version with sed
+      run: |
+        pushd nuxeo-web-ui-itests
+        grep -rl '10.10-HF[0-9]\{2\}-SNAPSHOT' . --exclude-dir={.git,node_modules,bower_components} | xargs sed -i 's/10.10-HF[0-9]\{2\}-SNAPSHOT/10.10-HF${{ github.event.inputs.hf_version }}/g'
```
Comment (Gabez0r) [inline .github/workflows/ondemand-main.yaml]: We could also try to filter for the files we are expecting to affect using `--include`. E.g.: `--include=**/pom.xml`.

**Review by Gabez0r — COMMENTED** [review]
Great job! 👏 💪  Did you try this on legacy QA with the seds done manually? Does it run?

File: .github/workflows/ondemand-main.yaml (hunk 2 above)
Comment (richardsd) [reply .github/workflows/ondemand-main.yaml:86]: You are totally right, we are missing the dataviz-elements in both the new and old pipelines. I'll add it, don't think it is that hard to add as well. Thanks

File: .github/actions/get-branch/action.yaml (hunk 1 above)
Comment (richardsd) [reply .github/actions/get-branch/action.yaml]: 🤔  You are partially right, I need to review the use of both `''` and `""`, but as a general rule, we need to use `""` in places where we need to perform variable substitution (which is not the case in this particular line).

File: .github/workflows/ondemand-main.yaml (hunk 3 above)
Comment (richardsd) [reply .github/workflows/ondemand-main.yaml]: We don't really need it. I'm doing this to keep each step as an atomic unit, but we can try to simplify the workflow.

File: .github/actions/get-branch/action.yaml (hunk 1 above)
Comment (richardsd) [reply .github/actions/get-branch/action.yaml]: I'm reviewing it 👍 Thanks

File: .github/workflows/ondemand-main.yaml (hunk 2 above)
Comment (richardsd) [reply .github/workflows/ondemand-main.yaml:86]: Done. Let me know what you think about it.

**Review by mnixo — APPROVED** [review]

**Comment by richardsd** [comment]
Closing this one in favor of https://github.com/nuxeo/nuxeo-web-ui/pull/1359.

**Pull request closed without merging**